app/models/grda_warehouse/warehouse_reports/time_homeless_for_exit.rb

```diff
@@ -145,13 +145,15 @@ def average_days
     end
 
     private def client_scope
-      apply_filters(client_source.viewable_by(filter.user))
+      apply_filters(client_source.destination_visible_to(filter.user))
     end
 
     private def apply_filters(scope)
       scope = scope.send(filter.sub_population).
         joins(source_enrollments: :project).
-        merge(GrdaWarehouse::Hud::Project.where(id: filter.effective_project_ids))
+        merge(GrdaWarehouse::Hud::Project.viewable_by(filter.user))
+
+      scope = scope.merge(GrdaWarehouse::Hud::Project.where(id: filter.effective_project_ids)) if filter.effective_project_ids.reject(&:zero?).any?
```

Comment:
Makes it possible to get results even if no effective project ids are selected

Reply:
what does that mean?  If you can't see any projects, you shouldn't see any data (at least I don't think so).

Reply:
Oh okay. So the user shouldn't be able to run the report without selecting at least 1 of Project, Project Group, Organization, or Data Source?

Reply:
Good question.  I guess what you have is good, limit to the projects a user can see, then if they've chosen any projects, further limit to those.  I was confused about the UI, but after looking that appears to be the desired process.